
# Investigating food production-associated DNA methylation changes in paleogenomes: lack of consistent signals beyond technical noise

  

### Replicated Datasets Generation

First, one should generate the MSs annotated datasets using the script below.

  
```
Rscript --verbose --vanilla process_and_filter_epiPALEOMIX.R
```

After that the output of these files are used in the script:

  
```
sbatch replicated_dataset_generation.sh /path/to/previous/outputs /output/directory/name /path/to/reference/annotation/file
```

This uses replicate.py script to generate the desired output for replicated dataset generation.

  

### Downsampled Datasets Generation

  

20 Downsampled datasets are generated using downsampled_dataset_generation.R

by navigating to the directory which includes generated replicated datasets per individual.

Note: the names *_replicated should be replaced by the actual file names generated when reproducing the results.

  
```
Rscript --verbose --vanilla downsampled_dataset_generation.R
```
After generating these files, one can merge all the individual files by using cat from the commandline (e.g. all_replicated_sorted_$i where i in [1,20]).

Then, generate a file that has the mean Methylation Scores per individual per gene using the script below.

  
```
sbatch reshape_replicated_dataset.sh
sbatch downsampled_means_per_gene.sh
```

Previous batch files call reshape_replicated_dataset.R and downsampled_means_per_gene.R script files, respectively.

  

### Analysis of Variance on Replicated Dataset

  

One can use the ANOVA.sh file to carry out the Analysis of Variance on our 20 downsampled datasets by simply using the command below.

  
```
sbatch ANOVA.sh
```

This batch file calls ANOVA.R script for all 20 downsampled replicated files.

  

### Matrix Generation

The second type of MS dataset aside from the replicated MS dataset is an annotated matrix with CpG positions and individuals is used for different analyses. The CpG site reference dataset is merged with the processed epiPALEOMIX output of every single individual and annotated. Missing MS values are represented by NA. The matrix is generated by using the script:

  
```
sbatch join_matrix.sh
```

### Plotting the Figures

  

Below script plots the Violin plots shown in Figure 1. It uses the generated matrix which is described in the previous section as input.

  
```
sbatch violin_plots_figure1B.sh
```

This batch script uses violin_plots_figure1B.R script and it requires the working directory of the generated matrix.


The scatter plots shown in Figure 1 can be replicated via running the next script. If one investigates the script, they will see a special type of input file is required for this script (CGIs). This file includes the genomic areas categorized merged with our MS dataset.

  
```
Rscript --verbose --vanilla mean_MS_per_individual_on_genomic_areas.R
```

 
In Figure 2, we see the plots of candidate genes for each category (subsistence type, tissue type and genetic sex) and for each model described in the Methods section of the article. The figure can be plotted using the following script.

  
```
Rscript --verbose --vanilla scatterplots_of_candidate_genes.R
```

The scripts inside this R file require the input files with the information related to the selected genes. They are basically the grepped positions from the generated matrix.

  

Multi-dimensional Scaling (Figure 3) analysis was carried out using the following file. It requires the means of all 20 downsampled datasets with respect to observed genes and information file related to the individuals. One can run the script using the command below.

```
Rscript --verbose --vanilla multi_dimensional_scaling.R
```

X-chromosome analysis just uses the normalized MSs and basic violin plots are realized after (Figure 4).

### Other analyses mentioned

All the other analyses mentioned in the article are carried out via using the scripts inside the general_analysis.R file.


For further information, please refer to the article.

The R scripts and batch files are assumed to run using a slurm workload manager. If this is not the case for your system, you can replace "Rscript --verbose --vanilla" with "R CMD BATCH" and after rearranging these, you can use "bash script-name-here.sh" in Linux.

A preview of all the datasets mentioned in the codes are provided under the "data" directory.